cmake/Modules/CodeCoverage.cmake

```diff
@@ -170,9 +170,10 @@ function(ADD_CODE_COVERAGE)
 
     # Create Python coverage report
     add_custom_target(${Coverage_NAME}_py
-        COMMAND cp ${PROJECT_BINARY_DIR}/.coverage ${COVERAGE_DIR}/.coverage.nosetests || echo "WARNING: No nosetest coverage!"
+        COMMAND cp ${PROJECT_BINARY_DIR}/.coverage ${COVERAGE_DIR}/.coverage.python || echo "WARNING: No python coverage!"
         COMMAND python-coverage combine || echo "WARNING: No python coverage to combine!"
-        COMMAND python-coverage xml || echo "WARNING: No python xml to output"
+        COMMAND python-coverage report --include "*${PROJECT_SOURCE_DIR}*" --omit ${COVERAGE_EXCLUDES}
+        COMMAND python-coverage html --include "*${PROJECT_SOURCE_DIR}*"  --omit ${COVERAGE_EXCLUDES} || echo "WARNING: No python html to output"
```

Comment:
Please either a) make the HTML an option or b) generate both XML and HTML. I know of several projects that are using codecov.io to process the XML output, so we can't break that workflow.

I presume we should add the include/omit to the XML report generation as well though?

Reply:
Thanks for the feedback. I opened a new PR #24 which, in combination with PR #23, splits the different changes into smaller commits. Also I now only add things.